jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -98,17 +97,34 @@ public ResourceCollection(Resource... resources)
      */
     public ResourceCollection(String[] resources)
     {
-        _resources = new Resource[resources.length];
+        if(resources == null || resources.length ==0)
+        {
+            _resources = null;
+            return;
+        }
+
+        ArrayList<Resource> res = new ArrayList<>();
+
         try
         {
-            for(int i=0; i<resources.length; i++)
+            for(String strResource: resources)
+            {
+                if(strResource == null || strResource.length() == 0)
+                    continue; // skip null and empty string only (whitespace only is valid)
```

Comment:
I think if we're going to start testing the resource array elements, then we shouldn't accept null entries, we should throw IAE. Probably same for empty and all whitespace entries.

Reply:
Whitespace only paths are a real thing.  So are paths that start/end with whitespace (so no trimming)

Reply:
We were very inconsistent.  

For example, CSV wouldn't ISE/IAE (just log a warn)

Reply:
But this isn't a path like in the call to addPath(String path), this is adding a Resource that should have a location on disk. I can't see how a null location is wrong but an empty or whitespaceonly location is ok?